lib/rets/metadata/file_cache.rb

```diff
@@ -0,0 +1,29 @@
+module Rets
+  module Metadata
+
+    # This metadata cache persists the metadata to a file.
+    class FileCache
+
+      def initialize(path)
+        @path = path
+      end
+
+      # Save the metadata.
+      def save(metadata)
+        File.open(@path, "wb") do |file|
+          Marshal.dump(metadata, file)
+        end
+      end
+
+      # Load the metadata.  Return nil if it could not be loaded.
+      def load
+        File.open(@path, "rb") do |file|
+          Marshal.load(file)
+        end
+      rescue IOError, SystemCallError
+        nil
+      end
```

Comment:
Ah I did not realize how marshal worked (it seems so obvious now).

I guess I dislike `Marshall` it purely because it involves mutation to add the logger onto the `Metadata::Root` object. Marshal could be an implementation detail of `FileCache` and we could persist sources upon save and initialize a `Metadata::Root` with sources and logger on load (the code I put above but with `Marshal.dump(metadata.sources, file)`).

That said I think this represents an improvement on what we currently have.

I would welcome other people's opinions on whether we should tie ourselves to Marshal or not. Whilst it's pretty minor it feels like something we should get a consensus on before we merge and commit to one or the other.